### PR TITLE
APP-2541: make visibility required for cli create command

### PR DESCRIPTION
**Pull request by RoxyFarhad**
This change should make it so that when you upload your meta.json file there must be visibility set in the file. 

**OLD - no longer relevant**

Creating a private module in staging:
Screenshot 2023-08-29 at 11 25 46 AM

Creating a public module in staging:
Screenshot 2023-08-29 at 11 26 39 AM

Bad Value:
Screenshot 2023-08-29 at 11 27 49 AM

No Value:
Screenshot 2023-08-29 at 11 28 21 AM

### CODE REVIEW

**Review by zaporter-work — APPROVED** [review]

**Comment by Fahmina** [comment]
Hey @RoxyFarhad -- Sorry if we miscommunicated. The intention of the ticket was to change it so that the meta.json file created after a user does `viam module create` does not autopopulate visibility to "private". 
We DO NOT want to add a CLI arg for visibility.

**Comment by RoxyFarhad** [comment]
> Sorry if we miscommunicated. The intention of the ticket was to change it so that the meta.json file created after a user does viam module create does not autopopulate visibility to "private".
We DO NOT want to add a CLI arg for visibility.

Ah okay, I got that we didn't want to explicitly make it private. But I assumed we would then want to explicitly set a field on create. So will make it so the client doesn't require an arg, but when the metadata file is uploaded it needs to have a visibility key set? 

IMO i feel like its nicer to set it on create or update via CLI as finding and changing meta.json file is little more work but happy to revert that anyways.

File: cli/module_registry.go [hunk 1]
```diff
@@ -144,6 +144,10 @@ func UpdateModuleAction(c *cli.Context) error {
 		}
 	}
 
+	if manifest.Visibility != moduleVisibilityPrivate && manifest.Visibility != moduleVisibilityPublic {
+		return errors.Errorf("manifest visibility must be either 'private' or 'public' but you provided %s", manifest.Visibility)
```
Comment (RoxyFarhad) [inline cli/module_registry.go:148]: if this is never changed, it will be an empty string which will fail @Fahmina

Comment (zaporter-work) [reply cli/module_registry.go:148]: I think that this error case is already handled.. not bad to check again though
```
➜  cli git:(zp/2488) ✗ ./main module update
Error: invalid module visibility. must be either "public" or "private"
```

**Review by zaporter-work — APPROVED** [review]
LGTM!

**Comment by RoxyFarhad** [comment]
> I think that this error case is already handled.. not bad to check again though

Ah missed that, removed the additional check then

**Review by michaellee1019 — APPROVED** [review]

**Pull request merged**